src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowLoopSpec.groovy

```diff
@@ -0,0 +1,618 @@
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.statusCode == HttpStatus.BAD_REQUEST
+        with(exc.responseBodyAsString.to(MessageError)) {
+            errorMessage == "Could not update flow"
+            errorDescription == "Source switch $switchPair.src.dpId is not connected to the controller"
+        }
+
+        and: "FlowLoop is not created"
+        with(northbound.getFlow(flow.flowId)) {
+            !it.looped
+        }
+
+        when: "Try to create flowLoop on th dst switch(activated)"
+        northboundV2.createFlowLoop(flow.flowId, new FlowLoopPayload(switchPair.dst.dpId))
+
+        then: "Human readable error is returned" //system can't update the flow when it is down
+        def exc2 = thrown(HttpClientErrorException)
+        exc2.statusCode == HttpStatus.BAD_REQUEST
+        with(exc2.responseBodyAsString.to(MessageError)) {
+            errorMessage == "Could not update flow"
+            errorDescription == "Source switch $switchPair.src.dpId is not connected to the controller"
+        }
+
+        then: "FlowLoop is not created"
+        with(northbound.getFlow(flow.flowId)) {
+            !it.looped
+        }
+
+        and: "FlowLoop rules are not created on the dst switch"
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            assert northbound.getSwitchRules(switchPair.dst.dpId).flowEntries.findAll {
+                new Cookie(it.cookie).getType() == CookieType.FLOW_LOOP
+            }.empty
+        }
+
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        switchHelper.reviveSwitch(switchPair.src, blockData, true)
+    }
+
+    @Tidy
+    @Tags(LOW_PRIORITY)
+    def "Unable to create flowLoop on the src switch when it is already created on the dst switch"() {
+        given: "An active flow with created flowLoop on the src switch"
+        def switchPair = topologyHelper.getNeighboringSwitchPair()
+        def flow = flowHelperV2.randomFlow(switchPair)
+        flowHelperV2.addFlow(flow)
+        northboundV2.createFlowLoop(flow.flowId, new FlowLoopPayload(switchPair.src.dpId))
+
+        when: "Try to create flowLoop on the dst switch"
+        northboundV2.createFlowLoop(flow.flowId, new FlowLoopPayload(switchPair.dst.dpId))
+
+        then: "FlowLoop is not created on the dst switch"
+        def exc2 = thrown(HttpClientErrorException)
+        exc2.statusCode == HttpStatus.BAD_REQUEST
+        with(exc2.responseBodyAsString.to(MessageError)) {
+            errorMessage == "Could not update flow"
+            errorDescription == "Can't change loop switch"
+        }
+
+        cleanup:
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+    }
+
+    @Tidy
+    @Tags(LOW_PRIORITY)
+    def "Unable to create flowLoop on a transit switch"() {
+        given: "An active multi switch flow with transit switch"
+        def switchPair = topologyHelper.getNotNeighboringSwitchPair()
```

Comment:
Fine for now, but keep in mind that we have a smaller amount of switches on stage now, and may want to pick neighboring and manipulate flow path via costs at some point. This sound like a separate task, we may want some easier tools for that to hide such code from every test in the future